packages/admin/admin/src/common/DialogTitle.tsx

```diff
@@ -0,0 +1,75 @@
+import { Close } from "@comet/admin-icons";
+import { ComponentsOverrides, css, DialogTitle as MUIDialogTitle, IconButton as MUIIconButton, Theme, useThemeProps } from "@mui/material";
+import { ReactNode } from "react";
+
+import { createComponentSlot } from "../helpers/createComponentSlot";
+import { ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+
+export type DialogTitleClassKey = "dialogTitle" | "iconButton" | "contentWrapper";
+
+export interface DialogTitleProps
+    extends ThemedComponentBaseProps<{
+        iconButton: typeof MUIIconButton;
+        dialogTitle: typeof MUIDialogTitle;
+        contentWrapper: "div";
+    }> {
+    children?: ReactNode;
+    onClose?: () => void;
+    iconMapping?: {
+        closeIcon?: React.ReactNode;
+    };
+}
+export function DialogTitle(inProps: DialogTitleProps) {
+    const { slotProps, iconMapping = {}, children, onClose, ...restProps } = useThemeProps({ props: inProps, name: "CometAdminDialogTitle" });
+    const { closeIcon = <Close color="inherit" /> } = iconMapping;
+    return (
+        <Title {...slotProps?.dialogTitle} {...restProps}>
+            <ContentWrapper {...slotProps?.contentWrapper}>{children}</ContentWrapper>
+            {onClose && (
+                <IconButton {...slotProps?.iconButton} onClick={onClose}>
+                    {closeIcon}
+                </IconButton>
+            )}
+        </Title>
+    );
+}
+
+const Title = createComponentSlot(MUIDialogTitle)<DialogTitleClassKey>({
+    componentName: "DialogTitle",
+    slotName: "dialogTitle",
+})(css`
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+`);
+
+const IconButton = createComponentSlot(MUIIconButton)<DialogTitleClassKey>({
+    componentName: "DialogTitle",
+    slotName: "iconButton",
+})(css`
+    color: inherit;
+`);
```

Comment:
Please make sure the button does not change the height of the `DialogTitle`

| With `CloseButton` | Without `CloseButton` |
| --- | --- |
| <img width="705" alt="Dialog with close button" src="https://github.com/user-attachments/assets/b02282fd-a268-4b56-88c5-54656f425c93"> | <img width="705" alt="Dialog without close button" src="https://github.com/user-attachments/assets/7047d7ca-e0f8-4f14-a5c3-833ab5d77935"> |